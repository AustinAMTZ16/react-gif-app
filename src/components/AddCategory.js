import React, {useState} from 'react';
import ProTypes from 'prop-types';

export const AddCategory = ({estableceCategorias}) => {
    const [inputValue, setInputValue] = useState('')

    const hadleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const hadleSubmit = (e) =>{
        e.preventDefault();
        console.log('Submit hecho');
        if(inputValue.trim().length > 2){
            estableceCategorias(categorias =>[inputValue, ...categorias]);
            setInputValue('');
        }        
    }
    return (
        <>
            <form onSubmit={hadleSubmit}>
                <h3>Buscar: {inputValue}</h3>
                <input 
                    type="text"
                    value={ inputValue }
                    onChange={hadleInputChange}
                />
            </form>
        </>
    )
}

AddCategory.prototype={
    estableceCategorias: ProTypes.func.isRequired
}