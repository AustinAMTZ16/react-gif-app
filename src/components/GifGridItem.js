import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({id, title, url}) => {
    //console.log({id, title, url});
    return (
        <div className="card animate__animated animate__bounce">
            <img src={url} alt={title}/>
            <p>{title}</p>             
            <p>{id}</p>
        </div>
    )
}

GifGridItem.prototype={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}