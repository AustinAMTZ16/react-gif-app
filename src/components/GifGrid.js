import React, {useState, useEffect} from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({categoria})=>{
    
    const {data:images, loading} = useFetchGifs(categoria);
    /*const [images, setImages] = useState([]);

    useEffect( () => {
        getGifs(categoria)
            .then(setImages);
    }, [categoria])*/

    return(
        <>
            <h3>{categoria}</h3>  
            {loading && <p className="animate__animated animate__flash">Cargando imagenes...</p>}
            <div className="card-grid">                  
                {
                    images.map(img => (
                        <GifGridItem 
                            key={img.id}
                            //img={img} 
                            {...img}
                        />
                    ))
                }            
            </div>
        </>        
    )
}