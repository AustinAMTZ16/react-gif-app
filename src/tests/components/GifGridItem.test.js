import { shallow } from 'enzyme';
import React from 'react'
import {GifGridItem} from '../../components/GifGridItem'

describe('Prueba <GifGridItem />', () => {
    const title = 'title';
    const url = 'https://gif.dck.com.mx/';
    const wrapper = shallow(<GifGridItem 
            title = {title}
            url = {url}
        />);
    test('<GifGridItem /> debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    }) 
    test('debe de tener un pÃ¡rrafo con el title', () => {
       /* const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);*/
    })
    test('debe de tener la imagen igua al url y alt de los props', () => {
        const img = wrapper.find('img');
        console.log(img.props());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    test('debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        console.log(div.prop('className'))
        const className = div.prop('className');
        expect(className.includes('animate__bounce')).toBe(true)
    })
    
})
