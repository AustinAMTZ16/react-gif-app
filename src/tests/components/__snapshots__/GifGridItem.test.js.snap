// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifGridItem /> <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="title"
    src="https://gif.dck.com.mx/"
  />
  <p>
    title
  </p>
  <p />
</div>
`;
